src/sciline/pipeline.py

```diff
@@ -153,10 +300,48 @@ def __setitem__(self, key: Type[T], param: T) -> None:
             )
         self._set_provider(key, lambda: param)
 
-    def _set_provider(self, key: Type[T], provider: Callable[..., T]) -> None:
+    def set_param_table(self, params: ParamTable) -> None:
+        """
+        Set a parameter table for a row dimension.
+
+        Values in the parameter table provide concrete values for a type given by the
+        respective column header.
+
+        A pipeline can have multiple parameter tables, but only one per row dimension.
+        Column names must be unique across all parameter tables.
+
+        Parameters
+        ----------
+        params:
+            Parameter table to set.
+        """
+        if params.row_dim in self._param_tables:
+            raise ValueError(f'Parameter table for {params.row_dim} already set')
```

Comment:
What if a user wants to replace a table? Is this a reasonable thing to want to do?

Reply:
We are currently not allowing this for providers and params either. Could be added in the future, but maybe not worth the complexity.